src/Collection.php

```diff
@@ -79,4 +80,14 @@ public function removeElement(mixed $element);
      * @return void
      */
     public function set(string|int $key, mixed $value);
+
+    /**
+     * {@inheritDoc}
+
+     * @return Collection<mixed>[] An array with two elements. The first element contains the collection
```

Comment:
Using `Collection<mixed>` here does not make sense. Tools supporting generics should see `Collection<TKey, T>`.

Reply:
As this is a copy/paste from the docs of the ReadableCollection I'd then also fix that there...

And also fix the conflict

Reply:
After checking with the sources: So far the `@return` annotations all contain references to `mixed`. Should this then be changed everywhere?

Personally I'd use solely `@return array{0: Collection<TKey, T>, 1: Collection<TKey, T>}` and skip the `@return-psalm` annotation...

Reply:
I agree, these annotations are consistent with `ReadableCollection::partition()`. Merging `@return` and `@psalm-return` is probably out of scope here.